packages/core/src/components/templates/ProductListingPage/ProductListingPage.tsx

```diff
@@ -70,69 +81,95 @@ const useSearchParams = ({
   return useMemo(() => parseSearchState(new URL(hrefState)), [hrefState])
 }
 
+const overwriteMerge = (_, sourceArray, __) => sourceArray
+
 export default function ProductListingPage({
   page: { sections, settings },
-  ...otherProps
+  data: server,
 }: ProductListingPageProps) {
-  const { collection } = otherProps
+  const collection = server.collection
   const router = useRouter()
   const applySearchState = useApplySearchState()
   const searchParams = useSearchParams({
-    ...otherProps,
+    collection,
     sort: settings?.productGallery?.sortBySelection as SearchState['sort'],
   })
 
-  const { page, sort } = searchParams
+  const { page, sort, term, selectedFacets } = searchParams
   const title = collection?.seo.title ?? storeConfig.seo.title
   const description = collection?.seo.description ?? storeConfig.seo.title
   const pageQuery = page !== 0 ? `?page=${page}` : ''
   const separator = pageQuery !== '' ? '&' : '?'
   const sortQuery = !!sort ? `${separator}sort=${sort}` : ''
   const [pathname] = router.asPath.split('?')
   const canonical = `${storeConfig.storeUrl}${pathname}${pageQuery}${sortQuery}`
+  const itemsPerPage = settings?.productGallery?.itemsPerPage ?? ITEMS_PER_PAGE
+  const initialPage = useRef(page)
+
+  const { data: pageProductGalleryData } = useProductGalleryQuery({
+    term,
+    sort,
+    selectedFacets,
+    itemsPerPage,
+  })
+
+  const { productsPerPage } = usePageProductsQuery({
+    page: initialPage.current,
+    term,
+    sort,
+    selectedFacets,
+    itemsPerPage,
+  })
+
+  const context = {
+    data: {
+      ...deepmerge(
+        { ...server },
```

Comment:
```suggestion
        { ...serverData },
```

Reply:
`data.serverData` doesn't look good! Keeping it as `server` is cleaner.

Reply:
I missed where are are calling `data.server` 👀

Reply:
Alright, I realized my stupidity now. I dislike server because we're not really returning a... server here, right? What does the server data contain? Can't we name the variable as that?

Reply:
resolving this conversation because we already talked about it :)